rfcs/0000-changes-lifecyle.md

```diff
@@ -0,0 +1,205 @@
+- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [FuelLabs/sway-rfcs#0000](https://github.com/FuelLabs/sway-rfcs/pull/001)
+- Sway Issue: [FueLabs/sway#0000](https://github.com/FuelLabs/sway/issues/001)
+
+# Summary
+
+[summary]: #summary
+
+This RFC intends to propose a scheme to track changes in the compiler. 
+
+# Motivation
+
+[motivation]: #motivation
+
+The compiler needs a guide on how to introduce changes to improve developer experience every time a compiler is updated.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+## PR changes
+
+Every PR will have a section named "Release Notes", which will contains a user friendly explanation of what the PR contains.
```

Comment:
We should probably introduce the rest of the changeset benefits, including specifying the level of change the PR does (minor, major, patch, etc).

Reply:
Have we considered putting the release notes into a changelog file, rather than having them spread across different PR descriptions?

Reply:
For the moment I moved this to the "open questions" section.

Reply:
Another option is to generate the changelog file from commits as part of the release process, using something like https://git-cliff.org/

We'd have to be more rigorous about commit messages, i.e. using `feat!:` to indicate a breaking change for a feature.